pkg/operator/controller/garden/care/reconciler.go

```diff
@@ -66,6 +69,14 @@ func (r *Reconciler) Reconcile(reconcileCtx context.Context, req reconcile.Reque
 		return reconcile.Result{}, fmt.Errorf("error retrieving object from store: %w", err)
 	}
 
+	if !r.managedResourceWatchRegistered && r.registerManagedResourceWatchFunc != nil {
+		if err := r.registerManagedResourceWatchFunc(); err != nil {
+			log.Error(err, "Failed registering watch for resources.gardener.cloud/v1alpha1.ManagedResource now that a operator.gardener.cloud/v1alpha1.Garden object has been created")
```

Comment:
What is the reasoning of not propagating the error? Do we simply expect this to fail initially and succeed eventually?

What happens if the `Garden` is deleted? Will this also trigger the same crash as the watch is registered?

Reply:
> What is the reasoning of not propagating the error? Do we simply expect this to fail initially and succeed eventually?

Watching `ManagedResource`s is not essential and just an optimization to improve the "speed of reaction" in the system. Hence, I decided to just log it and not return the error via the reconciler (which prevents all health checks just because this optimized watch cannot be registered). Overall, I think this is a theoretical discussion anyways.

> What happens if the `Garden` is deleted? Will this also trigger the same crash as the watch is registered?

Good question, maybe 😄 I didn't test this scenario. The `ManagedResource` CRD will be deleted as part of `Garden` deletion, but since it existed before, maybe this `garden-care` controller can continue to run.
Anyways, it's a much less interesting problem since after deleting a `Garden` people will probably also delete `gardener-operator` itself (why else would you delete the `Garden`).
This change focuses on the other direction (deploying `gardener-operator` and creating a `Garden` for the first time) which is the much more common scenario, I assume :)

Reply:
The health checks will fail anyway directly at the beginning, if there is no ManagedResource CRD ([ref](https://github.com/gardener/gardener/blob/34ddd4c42589307eba0d7a7f0cbd709387c6c07a/pkg/operator/controller/garden/care/health.go#L74-L81)).
The garden conditions might even show the error.

Reply:
@oliver-goetz Yes, but I don't know what other reasons exist that could lead to an error when trying to register the watch.

Reply:
I'm fine with seeing the error in those garden conditions. That's what I wanted to say 😄

Reply:
@rfranzke Thanks for clarifying. I agree that stability of gardener operator after deletion of the `Garden` is probably not a very important case.

/lgtm
/unhold